### DISCUSSION TITLE
Updates from Mutation Responses

### DISCUSSION THREAD

**Discussion by Dave-Patsy** [TanStack/query#4878] [Q&A] [answered]
I am trying to make a mutation that appends the data on success to prevent a refetch . This is a noob problem :(

this is my fetch function and create component with useMutation.
![image](https://user-images.githubusercontent.com/24628331/214795528-776dae68-f7b2-4d52-875b-2b7224b5fa0b.png)

I see that it is calling the last ele in the dictionary data. I thought I was spreading the old state and appending the new task.
Should I just use the push function to append the new data to the oldData state and return that? 
![image](https://user-images.githubusercontent.com/24628331/214792861-49c53485-ed87-43bd-a2c9-11aba3f682a3.png)

**Comment by louis-young** [comment, marked as answer]
Hi 👋 

To answer the question: no, you probably shouldn't use `Array.prototype.push`. It looks like you're dealing with an object type. Also, updates to the query cache have to be made immutably. 

I'm not sure exactly what it is that you're asking for help with, but it looks like you might need to spread `data` into the new object you're setting into the cache. 

If you need any more help, please could you clarify your question so I can try and help?
